jablib/src/test/java/org/jabref/model/strings/StringUtilTest.java

```diff
@@ -433,4 +434,25 @@ void alignStringTable() {
 
         assertEquals(expected, StringUtil.alignStringTable(given));
     }
+
+    @ParameterizedTest(name = "{index}: normalize(\"{0}\") = \"{1}\"")
+    @MethodSource("provideNormalizationCases")
+    @DisplayName("Should normalize input strings as expected")
+    void testNormalize(String input, String expected) {
+        assertEquals(expected, StringUtil.normalize(input));
+    }
+
+    static Stream<org.junit.jupiter.params.provider.Arguments> provideNormalizationCases() {
+        return Stream.of(
+                Arguments.of(null, ""),
+                Arguments.of("", ""),
+                Arguments.of("   Hello   World  ", "Hello World"),
+                Arguments.of("apple  ,banana", "apple, banana"),
```

Comment:
I think, its OK if all spaces are removed... But I think, the logic could be more complexly intended?

I think, we don't need it that complex.

We could also switch to some edit distance algorithms...

Reply:
Either it works as expected or it does not. Accepting edge cases not working is misleading other devs, that maybe want to use the method too. See Java by comparison pp. 110-111. 😄

Reply:
Java by comparison assumes endless time to finish a feature.